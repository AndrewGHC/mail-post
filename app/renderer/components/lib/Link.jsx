// @flow
import * as React from 'react';
import styled from 'styled-components';
import { Link as RouterLink } from 'react-router-dom';

const StyledLink = styled(RouterLink)`
    text-decoration: none;
    &:focus, &:hover, &:visited, &:link, &:active {
        text-decoration: none;
    }
`;

type Props = {
  children: React.Node
};

const Link = ({ children, ...rest }: Props) => (
  <StyledLink {...rest}>
    {children}
  </StyledLink>
);

export default Link;
