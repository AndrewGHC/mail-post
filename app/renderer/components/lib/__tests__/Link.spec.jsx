import React from 'react';
import { shallow } from 'enzyme';

import Link from '../Link';

describe('Link', () => {
  it('renders a Link and passes props', async () => {
    const wrapper = shallow(<Link />);
    expect(wrapper).toMatchSnapshot();
  });
});
