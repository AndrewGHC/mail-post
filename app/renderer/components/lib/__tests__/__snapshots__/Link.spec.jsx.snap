// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders a Link and passes props 1`] = `<Link__StyledLink />`;
